### PR TITLE
lib: Add a way to override rustfmt path.

### PR DESCRIPTION
**Pull request by emilio**
I'll need it to format some stuff on mozilla-central.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -601,6 +601,12 @@ impl Builder {
         self
     }
 
+    /// Sets an explicit path to rustfmt, to use when rustfmt is enabled.
+    pub fn with_rustfmt<P: Into<PathBuf>>(mut self, path: P) -> Self {
```
Comment (fitzgen) [inline src/lib.rs]: Put this next to the other `rustfmt` related methods/options?

Comment (emilio) [reply src/lib.rs]: Good idea, done!

**Comment by emilio** [comment]
@bors-servo r=fitzgen

 * Always forget to update the lockfile.

**Pull request merged**